Resources/public/js/apps/ez-platformuiapp.js

```diff
@@ -140,9 +150,13 @@ YUI.add('ez-platformuiapp', function (Y) {
 
             this.on({
                 'loadingChange': this._loading,
-                'navigate': function () {
+                '*:navigate': function () {
                     this.set('loading', true);
-                }
+                },
+            });
+            this.after('routingEnabledChange', function () {
+                this._handleHash();
```

Comment:
I would rather add the condition on the new value of `routingEnabled` here to have better method names, so `_handleHash` would be split into `_preventHashChange` and `_allowHashChange` and `_alterRoutesCallback` into `_disableRouting` and `_enableRouting` for instance.